File: apps/mobile/ios/Visvine/Models/Directory.swift
import Foundation

/// Directory member — a Node row from /api/data/nodes, reused for /api/profile/[id].
/// Mirrors src/types/index.ts `DirectoryMember` (note snake_case wire fields).
struct DirectoryMember: Codable, Identifiable {
    let id: String
    var name: String
    var type: String
    var subtitle: String?
    var imageUrl: String?
    var location: String?
    var tags: [String]?
    var communityId: String?
    var title: String?
    var company: String?
    var email: String?

    enum CodingKeys: String, CodingKey {
        case id, name, type, subtitle, location, tags, title, company, email
        case imageUrl = "image_url"
        case communityId = "community_id"
    }
}

struct NodesResponse: Codable {
    let nodes: [DirectoryMember]
}

/// Partial profile patch sent by the edit screen (PATCH /api/profile/[id]).
struct ProfileUpdate: Codable {
    var name: String?
    var title: String?
    var company: String?
    var location: String?
}

File: apps/mobile/android/app/src/main/java/com/visvine/mobile/data/model/Directory.kt
package com.visvine.mobile.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Directory member — a Node row from /api/data/nodes, also reused for
 * /api/profile/[id]. Mirrors src/types/index.ts `DirectoryMember`. Note the
 * snake_case wire fields (`image_url`, `community_id`).
 */
@Serializable
data class DirectoryMember(
    val id: String,
    val name: String = "",
    val type: String = "person",
    val subtitle: String? = null,
    @SerialName("image_url") val imageUrl: String? = null,
    val location: String? = null,
    val tags: List<String>? = null,
    @SerialName("community_id") val communityId: String? = null,
    val title: String? = null,
    val company: String? = null,
    val email: String? = null,
)

@Serializable
data class NodesResponse(
    val nodes: List<DirectoryMember> = emptyList(),
)

/** Partial profile patch sent by the edit screen (PATCH /api/profile/[id]). */
@Serializable
data class ProfileUpdate(
    val name: String? = null,
    val title: String? = null,
    val company: String? = null,
    val location: String? = null,
)
